autotest/gcore/cog.py

```diff
@@ -586,3 +586,59 @@ def test_cog_invalidation_by_metadata_change():
 
     with gdaltest.error_handler():
         gdal.GetDriverByName('GTiff').Delete(filename)
+
+
+###############################################################################
+# Test a tiling scheme with a CRS with northing/easting axis order
+# and non power-of-two ratios of scales.
+
+
+def test_cog_northing_easting_and_non_power_of_two_ratios():
+
+    filename = '/vsimem/cog.tif'
+    ds = gdal.Translate(filename, 'data/byte.tif',
+        options='-of COG -a_srs EPSG:2193 -a_ullr 1000001 5000001 1000006.6 4999995.4 -co TILING_SCHEME=NZTM2000 -co ALIGNED_LEVELS=2')
+    assert ds.RasterXSize == 1280
+    assert ds.RasterYSize == 1280
+    b = ds.GetRasterBand(1)
+    assert [(b.GetOverview(i).XSize, b.GetOverview(i).YSize) for i in range(b.GetOverviewCount())] == [(512, 512), (256, 256)]
+
+    gt = ds.GetGeoTransform()
+    # 0.28 m/pixel corresponds to OGC ScaleDenom = 1000 / zoom level=14 in NZTM2000
+    assert gt == pytest.approx((999872, 0.28, 0, 5000320, 0, -0.28), abs=1e-8)
+
+    # Check that gt origin matches the corner of a tile at zoom 14
```

Comment:
Maybe I'm missing something, but how is the origin match being done in this test? I only see the tile_x/tile_y being calculated, but not the actual origin value being explicitly checked?

Reply:
it is checked by `assert gt == pytest.approx((999872, 0.28, 0, 5000320, 0, -0.28), abs=1e-8)` : (999872, 5000320) is the coordinate of the top-left corner

Reply:
Thank you @rouault. The variable naming makes this clearer for me know.